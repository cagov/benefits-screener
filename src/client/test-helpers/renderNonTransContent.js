import React from "react";
import { shallow } from "enzyme";
import { I18nextProvider, Trans } from "react-i18next";
import i18n from "./i18nForTests";
import shallowUntilTarget from "./shallowUntilTarget";

function renderNonTransContent(Component, props = {}) {
  const providersWrapper = shallow(
    <I18nextProvider i18n={i18n}>
      <Component {...props} />
    </I18nextProvider>
  );

  const wrapper = shallowUntilTarget(providersWrapper, Component.name);
  // Incorrectly using renderNonTransContent to render a component that
  // contains a Trans component results in a silent failure (the rendered
  // snapshot will contain the Trans component itself instead of the rendered
  // contents). This test makes that silent failure more obvious.
  expect(wrapper.containsMatchingElement(<Trans />)).toBe(false);

  return wrapper;
}

module.exports = renderNonTransContent;
