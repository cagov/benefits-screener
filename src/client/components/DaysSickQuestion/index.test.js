import renderNonTransContent from "../../test-helpers/renderNonTransContent";
import Component from "./index";

describe("<DaysSickQuestion />", () => {
  it("renders the component", async () => {
    const wrapper = renderNonTransContent(Component, "DaysSickQuestion", {
      onChange: () => {},
    });

    expect(wrapper).toMatchSnapshot();
  });
});
