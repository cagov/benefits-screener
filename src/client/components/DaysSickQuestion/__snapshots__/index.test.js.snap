// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DaysSickQuestion /> renders the component 1`] = `
<FormGroup
  className="follow-up"
  controlId="days-sick"
>
  <FormLabel
    column={false}
    srOnly={false}
  >
    If Yes, enter the number of days (1 - 7) you were unable to work.
  </FormLabel>
  <FormText
    className="help-text"
  >
    You must report the numbers of days that you could not work during this week.
  </FormText>
  <FormControl
    maxLength={1}
    onChange={[Function]}
    pattern="[1234567]"
    required={true}
    style={
      Object {
        "width": "6rem",
      }
    }
    type="text"
    value=""
  />
  <Feedback
    type="invalid"
  >
    This field is required.
  </Feedback>
</FormGroup>
`;
